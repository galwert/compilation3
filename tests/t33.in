int func1(int param1)
{
    return 1;
}
int func2(int param1, int param2)
{
    return func1(param1) * param2;
}

void main()
{
    int x1 = func1(1);
    int x2 = func2(1,2);
    while (((false and false) and true) or true)
    {
        int y = 1;
        continue;
        break;
    }
    int x5;
}

int func3(int param1, int param2, bool param3)
{
    int param4 = func1(param3);   // Should fail here - wrong parameter type
    return 5;
}
int func4()
{
    return 1;
}
