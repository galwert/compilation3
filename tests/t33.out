---end scope---
param1 INT -1
---end scope---
param1 INT -1
param2 INT -2
---end scope---
y INT 2
---end scope---
---end scope---
x1 INT 0
x2 INT 1
x5 INT 2
line 25: prototype mismatch, function func1 expects arguments (INT)
